storybookjs/storybook
# React: Fix issue with react 18 implementation

Follow up on https://github.com/storybookjs/storybook/pull/19118

Issue: When implementing a rendering story that checks a new button isn't created when setting an arg, I realised that I had actually broken things in https://github.com/storybookjs/storybook/pull/18737.

Now we have this story, we shouldn't be able to break things in this way again 🤞 

## What I did

Changed the react callback behaviour

## How to test

1. Run rendering story
2. Also check that the `useEffect` helper still works (there's a `hooks.stories.ts` story for this)
3. The docs E2E should pick up if the callback isn't getting called properly.
